tests/integration/ef_tests/test_ef_blockchain_tests.py

```diff
@@ -79,11 +79,67 @@ async def setup_test_state(
             ).execute(caller_address=starknet_address)
 
             if not is_account_eoa(account):
+                # Write bytecode of contract
                 contract = get_contract_account(starknet_address)
                 await contract.write_bytecode(
                     hex_string_to_bytes_array(account["code"])
                 ).execute(caller_address=kakarot.contract_address)
 
+                # Write storage at correct values
+                # Get keys and values to store
+                storage_keys = list(account["storage"].keys())
+                storage_values = list(account["storage"].values())
+
+                # Split the integers into two felts
+                # Mask to extract the lower u128 part
+                mask_u128 = (1 << 128) - 1
+                ca_storage_entries = []
+
+                # Add an invariant: There must be as many storage keys as storage values
+                assert len(storage_keys) == len(storage_values)
+
+                for index in range(len(storage_keys)):
+                    # Key for storage is u256, it should be split into {high: u128, low: u128}
+                    key_high = int(storage_keys[index], 16) >> 128
+                    key_low = int(storage_keys[index], 16) & mask_u128
+
+                    # Value for storage is u256, it should be split into {high: u128, low: u128}
+                    value_high = int(storage_values[index], 16) >> 128
+                    value_low = int(storage_values[index], 16) & mask_u128
+
+                    # We append to the storage entries a tuple of (key: Cairo.U256, value: Cairo.U256)
+                    # Note that when serializing, we need to store low first and then high:
+                    # {low: u128, high: 128} => serde => [low, high]
+                    ca_storage_entries.append(
+                        ([key_low, key_high], [value_low, value_high])
+                    )
+
+                # Now we have to perform low level custom storage for Starknet
+                # Reference: https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/contract-storage/
+                for storage_key, storage_value in ca_storage_entries:
```

Comment:
do we need to reiterate on the `ca_storage_entries`? I feel like all this loop could be brought up to the `for index in range(len(storage_keys)):` loop.